Assets/MixedRealityToolkit/_Core/Services/SpatialAwarenessSystem/MixedRealitySpatialAwarenessSystem.cs

```diff
@@ -0,0 +1,289 @@
+
+        private GameObject surfaceParent = null;
+
+        /// <inheritdoc />
+        public GameObject SurfacesParent => surfaceParent != null ? surfaceParent : (surfaceParent = CreateSecondGenerationParent("Surfaces"));
+
+        /// <summary>
+        /// Creates a parent that is a child of the Spatial Awareness System parent so that the scene hierarchy does not get overly cluttered.
+        /// </summary>
+        /// <returns>
+        /// The <see cref="GameObject"/> to which spatial awareness objects will be parented.
+        /// </returns>
+        private GameObject CreateSecondGenerationParent(string name)
+        {
+            var secondGeneration = new GameObject(name);
+
+            secondGeneration.transform.parent = SpatialAwarenessRootParent.transform;
+
+            return secondGeneration;
+        }
+
+        #region IMixedRealitySpatialAwarenessSystem Implementation
+
+        /// <inheritdoc />
+        public HashSet<IMixedRealitySpatialObserverDataProvider> DetectedSpatialObservers { get; } = new HashSet<IMixedRealitySpatialObserverDataProvider>();
+
+        /// <inheritdoc />
+        public bool IsObserverRunning(IMixedRealitySpatialObserverDataProvider observer)
+        {
+            return false;
+        }
+
+        /// <inheritdoc />
+        public uint GenerateNewObserverId()
+        {
+            var newId = (uint)Random.Range(1, int.MaxValue);
+
+            foreach (var observer in DetectedSpatialObservers)
+            {
+                if (observer.SourceId == newId)
+                {
+                    return GenerateNewObserverId();
+                }
+            }
+
+            return newId;
+        }
+
+        /// <inheritdoc />
+        public void ResumeObserver(IMixedRealitySpatialObserverDataProvider observer)
+        {
+            foreach (var spatialObserver in DetectedSpatialObservers)
+            {
+                if (spatialObserver.SourceId == observer.SourceId)
+                {
+                    spatialObserver.StartObserving();
+                    break;
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public void SuspendObserver(IMixedRealitySpatialObserverDataProvider observer)
+        {
+            foreach (var spatialObserver in DetectedSpatialObservers)
+            {
+                if (spatialObserver.SourceId == observer.SourceId)
+                {
+                    spatialObserver.StopObserving();
+                    break;
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public void RaiseSpatialAwarenessObserverDetected(IMixedRealitySpatialObserverDataProvider observer)
+        {
+            DetectedSpatialObservers.Add(observer);
+        }
+
+        /// <inheritdoc />
+        public void RaiseSpatialAwarenessObserverLost(IMixedRealitySpatialObserverDataProvider observer)
```

Comment:
When would an observer be lost?

Reply:
If we unplug a hardware sensor like a depth camera, for example.